src/PlatformServices/Deployment/TestRunDirectories.cs

```diff
@@ -71,8 +74,45 @@ public string InMachineNameDirectory
         {
             get
             {
-                return Path.Combine(Path.Combine(this.RootDeploymentDirectory, DeploymentInDirectorySuffix), Environment.MachineName);
+                var machineName = GetMachineName();
+
+                return Path.Combine(Path.Combine(this.RootDeploymentDirectory, DeploymentInDirectorySuffix), machineName);
+            }
+        }
+
+        private static string _machineName = null;
+        private static string GetMachineName()
+        {
+            if (_machineName != null)
+            {
+                return _machineName;
+            }
+
+#if NETSTANDARD1_5_OR_GREATER
+            _machineName = Environment.MachineName;
+            
+#else
+            // Try getting the machine name using the API anyway,
+            var machineNameProperty = typeof(Environment).GetTypeInfo().DeclaredMembers.SingleOrDefault(p => p.Name == "MachineName") as PropertyInfo;
```

Comment:
Could we replace the string with a nameof call.

Reply:
We can't, `MachineName` doesn't exists in NETSTANDARD1_4.